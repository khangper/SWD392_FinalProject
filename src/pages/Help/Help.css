.Help-container-fluid {
    padding: 0 0px;
  }
  
  .explore-container {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    padding: 10px;
  }
  .explore-container .search-box {
    display: flex;
    align-items: center;
    padding: 6px 12px;  
    border: 2px solid #e9ecef;
    background-color: white;
    border-radius: 2px;
    width: 100%;
    height: 50px;
    margin-bottom: 20px;
  }
  
  .explore-container .search-box input {
    flex: 1;
    border: none;
    outline: none;
    background: transparent;
    font-size: 14px;
    padding: 6px 10px;
  }
  
  .explore-container .search-icon {
    width: 20px;
    height: 20px;
    margin-right: 10px;
  }
  .Help-grid-container-input {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
    gap: 15px;
  }
  
  .Help-grid-item-full {
    grid-column: span 12;
  }

  
  .Help-grid-center {
    display: flex;
    justify-content: center;
    margin-top: 50px;
  }
  
  .Help-grid-item-half {
    grid-column: span 6;
    width: 66%;
  }
  
  .help_stitle h2 {
    font-size: 30px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    color: #fff;
    text-align: center;
    margin-bottom: 30px;
  }
  
  .Help-explore_search {
    display: flex;
    justify-content: center;
    width: 100%;
  }
  
  .Help-explore_search .ui.input {
    width: 80%; /* Thay đổi kích thước chiều ngang của input tại đây */
  }
  
  .Help-srch_explore {
    margin-bottom: 20px;
    padding: 11px 20px;
    width: 100%;
    background: #fff !important;
    border: 1px solid #e5e5e5 !important;
    color: #333 !important;
    box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
  }
  
  .Help-srch_explore:focus {
    border: 1px solid #333 !important;
    box-shadow: none !important;
  }
  
  .Help_215v12 {
    float: left;
    width: 100%;
    padding: 80px 0;
    background: #333;
    position: relative;
    margin-bottom: 23px;
  }
  

  /* Define the grid container */
.Help1-Grid-container {
    display: grid;
    grid-template-columns: repeat(3, 1fr); /* Assuming a 3-column layout like col-md-4 */
    gap: 20px; /* Adjust the gap between grid items as needed */
  }
  
  /* Optional: Make the grid responsive */
  @media (max-width: 768px) {
    .Help1-Grid-container {
      grid-template-columns: repeat(2, 1fr); /* 2-column layout for tablets */
    }
  }
  
  @media (max-width: 576px) {
    .Help1-Grid-container {
      grid-template-columns: 1fr; /* 1-column layout for mobile */
    }
  }
  .value_props50 {
    background: #fff;
    border-radius: 3px;
    padding: 20px;
    border: 1px solid #efefef;
    box-shadow: none;
    flex: 1 1;
    float: left;
    width: 100%;
    margin-bottom: 30px;
    text-align: center;
    height: 200px;
    transition: transform 0.3s ease; /* Thêm hiệu ứng chuyển động */
}

.value_props50:hover {
    transform: translateY(-5px); /* Di chuyển phần tử lên trên khi hover */
}


.crse_content h3 {
  font-size: 20px;
  font-weight: 500;
  font-family: 'Roboto', sans-serif;
  text-align: left;
  margin-bottom: 15px;
  color: #333;
  margin-bottom: 37px;
  margin-top: 27px;
}


.value_icon img.Help-icon-page {
    width: 14%;
    height: 47px;
    object-fit: contain;

    font-size: 36px;
    margin-right: 0;
    color: #333;
    background: #ffecec;
    border-radius: 3px;
    padding: 10px 6px;
    display: inline-block;
}
.value_props51 {
  background: #fff;
  border-radius: 3px;
  border: 1px solid #efefef;
  padding: 20px;
  box-shadow: none;
  flex: 1 1;
  float: left;
  color: #333 !important;
  width: 100%;
  font-size: 14px;
  line-height: 26px;
  font-weight: 500;
  font-family: 'Roboto', sans-serif;
  margin-bottom: 30px;
  text-align: center;
  transition: background-color 0.3s ease, border-color 0.3s ease, color 0.3s ease; /* Thêm hiệu ứng chuyển động */
}

.value_props51:hover {
  background-color: #f0f0f0; /* Thay đổi màu nền khi hover */
  border-color: #ccc; /* Thay đổi đường viền khi hover */
  color: #555; /* Thay đổi màu chữ khi hover */
}


.Help2-grid-container {
  display: grid;
  grid-template-columns: repeat(3, 1fr); /* Mỗi hàng có 3 cột */
  gap: 20px;
}

.nav-tabs-help {
  display: flex;
  justify-content: center;
  list-style-type: none;
  padding: 0;
  margin-top: -21px;
}