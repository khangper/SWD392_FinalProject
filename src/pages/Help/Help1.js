import React, { useState } from 'react';
import "./Help.css"
import search_icon from '..//../assets/search.png';
import Wallet from '..//../assets/Wallet.png';
import Megaphone2 from '..//../assets/Megaphone2.png';
import Filecheckale from '..//../assets/Filecheckale.png';
import FileEdit from '..//../assets/FileEdit.png';
import CourseManagement from '..//../assets/CourseManagement.png';
import TrueFile from '..//../assets/TrueFile.png';
import Account from '..//../assets/Account.png';
import Destop from '..//../assets/Destop.png';
import BookHelp from '..//../assets/BookHelp.png';
import MobileHelp from '..//../assets/MobileHelp.png';



export default function Help() {
  const [activeTab, setActiveTab] = useState('instructor');

  const handleTabClick = (tab) => {
    setActiveTab(tab);
  };

  return (
    <div>
      <div className="Help-container-fluid">
        <div className="Help-grid-container-input">
          <div className="Help-grid-item-full">
           <div className="Help_215v12">
            <div className="Help-grid-center">
             <div className="Help-grid-item-half">
              <div className="help_stitle">
               <h2>How may we help you?</h2>
               <div className="explore_search">
            <div className="ui search focus">
                <div className="search-box flex-div">
                <img src={search_icon} alt='' className='search-icon-allInstructor' />
                    <input className="prompt srch_explore" type="text" placeholder="       Search Tutors..."/>
                    
                </div>
            </div>
        </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>


      <div className="Help_215b15">
        <div className=".Help-grid-container-input">
          <div className="Help-grid-item-full">
            <div className="col-lg-12">
              <div className="course_tabs">
                <nav>
                  <div className="nav nav-tabs-help help_tabs tab_crse justify-content-center" id="nav-tab" role="tablist">
                    <a
                      className={`tab ${activeTab === 'instructor' ? 'active' : ''}`}
                      id="nav-instructor-tab"
                      onClick={() => handleTabClick('instructor')}
                    >
                      Instructor
                    </a>
                    <a
                      className={`tab ${activeTab === 'student' ? 'active' : ''}`}
                      id="nav-student-tab"
                      onClick={() => handleTabClick('student')}
                    >
                      Student
                    </a>
                  </div>
                </nav>
              </div>
            </div>
          </div>
        </div>
      </div>

      <div className="_215b17">
        <div className="container-fluid">
          <div className="row">
            <div className="col-lg-12">
              <div className="course_tab_content">
                <div className="tab-content" id="nav-tabContent">
                  {activeTab === 'instructor' && (
                    <div className={`tab-pane fade ${activeTab === 'instructor' ? 'show active' : ''}`} id="nav-instructor" role="tabpanel">
<div className="tpc152">
  <div className="crse_content">
    <h3>Select a topic to search for help</h3>
  </div>
  <div className="section3126 mt-20">
    <div className="Help1-Grid-container">
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={Wallet} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Payments</h4>
            <p>Understand the revenue share and how to receive payments.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={Megaphone2} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Selling & Promotion</h4>
            <p>Learn about the announcement and promotional tools.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={Filecheckale} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Quality Standards</h4>
            <p>Learn what it takes to create a high quality course.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={FileEdit} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Course Building</h4>
            <p>Build your course curriculum and landing page.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={CourseManagement} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Course Management</h4>
            <p>Maintain your course and engage with students.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={TrueFile} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Trust & Safety</h4>
            <p>Policy and copyright questions and guidance.</p>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>

<div className="tpc152">
    <div className="crse_content">
        <h3>Frequently Asked Questions</h3>
    </div>
    <div className="section3126 mt-20">
        <div className="Help2-grid-container">
            <a href="#" className="value_props51">                                                                                                          
                Promote Your Course With Coupons and Referral Links                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Cursus Course Quality Checklist                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Instructor Revenue Share                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Instructor Promotional Agreements and Cursus Deals                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                How to Become an Instructor: FAQ                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                How to Select Your Payout Method And Become a Premium Instructor                                                                                                                    
            </a>
        </div>
    </div>
</div>
                    </div>
                  )}

                  {activeTab === 'student' && (
                    <div className={`tab-pane fade ${activeTab === 'student' ? 'show active' : ''}`} id="nav-student" role="tabpanel">
<div className="tpc152">
  <div className="crse_content">
    <h3>Select a topic to search for help</h3>
  </div>
  <div className="section3126 mt-20">
    <div className="Help1-Grid-container">
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
            <img src={Filecheckale} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Getting Started</h4>
            <p>Learn how Cursus works and how to start learning.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
            <img src={Account} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Account/Profile</h4>
            <p>Manage your account settings.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
            <img src={Destop} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Troubleshooting</h4>
            <p>Experiencing a bug? Check here.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
            <img src={BookHelp} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Course Taking</h4>
            <p>Everything about taking a course on Udemy.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
            <img src={Wallet} alt='' className='Help-icon-page' />
          </div>
          <div className="value_content">
            <h4>Purchase/Refunds</h4>
            <p>Learn about coupons, how to send gifts, and refunds.</p>
          </div>
        </a>
      </div>
      <div className="value_props50">
        <a href="#">
          <div className="value_icon">
          <img src={MobileHelp} alt='' className='Help-icon-page' />            
          </div>
          <div className="value_content">
            <h4>Mobile</h4>
            <p>On the go? Learn about our mobile app.</p>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>

<div className="Frequently-Quétion">
    <div className="crse_content">
        <h3>Frequently Asked Questions</h3>                                                                            
    </div>
    <div className="section3126 mt-20">
        <div className="Help2-grid-container">
            <a href="#" className="value_props51">                                                                                                          
                Lifetime Access                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Cursus FAQ                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Downloading Courses                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Certificate of Completion                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                Refund a Course                                                                                                                    
            </a>
            <a href="#" className="value_props51">                                                                                                          
                How to Solve Payment Issues                                                                                                                    
            </a>
        </div>
    </div>
</div>
	
                    </div>
                  )}
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
