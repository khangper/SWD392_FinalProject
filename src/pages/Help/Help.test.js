import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Help from "./Help1";

describe("Help Component", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  const renderWithRouter = (ui) => {
    return render(<Router>{ui}</Router>);
  };

  it("renders the Help component", () => {
    const { container } = renderWithRouter(<Help />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });

  it("displays the main title and search input", () => {
    renderWithRouter(<Help />);
    const titleElement = screen.getByText("How may we help you?");
    expect(titleElement).toBeInTheDocument();
    const searchInput = screen.getByPlaceholderText("Search Tutors");
    expect(searchInput).toBeInTheDocument();
  });

  it("renders the instructor tab as active by default", () => {
    renderWithRouter(<Help />);
    const instructorTab = screen.getByText("Instructor");
    const studentTab = screen.getByText("Student");
    expect(instructorTab).toHaveClass("active");
    expect(studentTab).not.toHaveClass("active");
  });

  it("switches to student tab when clicked", () => {
    renderWithRouter(<Help />);
    const instructorTab = screen.getByText("Instructor");
    const studentTab = screen.getByText("Student");
    fireEvent.click(studentTab);
    expect(instructorTab).not.toHaveClass("active");
    expect(studentTab).toHaveClass("active");
  });

  it("displays the correct content for the active tab", () => {
    renderWithRouter(<Help />);
    const instructorTab = screen.getByText("Instructor");
    const studentTab = screen.getByText("Student");

    // Check initial content for instructor tab
    expect(
      screen.getByText("Select a topic to search for help")
    ).toBeInTheDocument();
    expect(screen.getByText("Payments")).toBeInTheDocument();

    // Switch to student tab and check content
    fireEvent.click(studentTab);
    expect(
      screen.getByText("Select a topic to search for help")
    ).toBeInTheDocument();
    expect(screen.getByText("Getting Started")).toBeInTheDocument();
    expect(screen.queryByText("Payments")).not.toBeInTheDocument();
  });
});
