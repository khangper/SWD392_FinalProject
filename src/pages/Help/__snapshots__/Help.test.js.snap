// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Help Component renders the Help component 1`] = `
<div>
  <div>
    <div
      class="Help-container-fluid"
    >
      <div
        class="Help-grid-container-input"
      >
        <div
          class="Help-grid-item-full"
        >
          <div
            class="Help_215v12"
          >
            <div
              class="Help-grid-center"
            >
              <div
                class="Help-grid-item-half"
              >
                <div
                  class="help_stitle"
                >
                  <h2>
                    How may we help you?
                  </h2>
                  <div
                    class="explore_search"
                  >
                    <div
                      class="explore-container"
                    >
                      <section
                        class="search-box"
                      >
                        <img
                          alt="Search Icon"
                          class="search-icon"
                          src="search.png"
                        />
                        <input
                          placeholder="Search Instructor Help"
                          type="text"
                          value=""
                        />
                      </section>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Help_215b15"
    >
      <div
        class="Help-grid-container-input"
      >
        <div
          class="Help-grid-item-full"
        >
          <div
            class="col-lg-12"
          >
            <div
              class="course_tabs"
            >
              <nav>
                <div
                  class="nav nav-tabs-help help_tabs tab_crse justify-content-center"
                  id="nav-tab"
                  role="tablist"
                >
                  <a
                    class="tab active"
                    id="nav-instructor-tab"
                  >
                    Instructor
                  </a>
                  <a
                    class="tab "
                    id="nav-student-tab"
                  >
                    Student
                  </a>
                </div>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="_215b17"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-lg-12"
          >
            <div
              class="course_tab_content"
            >
              <div
                class="tab-content"
                id="nav-tabContent"
              >
                <div
                  class="tab-pane fade show active"
                  id="nav-instructor"
                  role="tabpanel"
                >
                  <div
                    class="tpc152"
                  >
                    <div
                      class="crse_content"
                    >
                      <h3>
                        Select a topic to search for help
                      </h3>
                    </div>
                    <div
                      class="section3126 mt-20"
                    >
                      <div
                        class="Help1-Grid-container"
                      >
                        <div
                          class="value_props50"
                        >
                          <a
                            href="/helpview1"
                          >
                            <div
                              class="value_icon"
                            >
                              <img
                                alt=""
                                class="Help-icon-page"
                                src="Wallet.png"
                              />
                            </div>
                            <div
                              class="value_content"
                            >
                              <h4>
                                Payments
                              </h4>
                              <p>
                                Understand the revenue share and how to receive payments.
                              </p>
                            </div>
                          </a>
                        </div>
                        <div
                          class="value_props50"
                        >
                          <a
                            href="/helpview2"
                          >
                            <div
                              class="value_icon"
                            >
                              <img
                                alt=""
                                class="Help-icon-page"
                                src="Megaphone2.png"
                              />
                            </div>
                            <div
                              class="value_content"
                            >
                              <h4>
                                Selling & Promotion
                              </h4>
                              <p>
                                Learn about the announcement and promotional tools.
                              </p>
                            </div>
                          </a>
                        </div>
                        <div
                          class="value_props50"
                        >
                          <a
                            href="/"
                          >
                            <div
                              class="value_icon"
                            >
                              <img
                                alt=""
                                class="Help-icon-page"
                                src="Filecheckale.png"
                              />
                            </div>
                            <div
                              class="value_content"
                            >
                              <h4>
                                Quality Standards
                              </h4>
                              <p>
                                Learn what it takes to create a high-quality course.
                              </p>
                            </div>
                          </a>
                        </div>
                        <div
                          class="value_props50"
                        >
                          <a
                            href="/"
                          >
                            <div
                              class="value_icon"
                            >
                              <img
                                alt=""
                                class="Help-icon-page"
                                src="FileEdit.png"
                              />
                            </div>
                            <div
                              class="value_content"
                            >
                              <h4>
                                Course Building
                              </h4>
                              <p>
                                Build your course curriculum and landing page.
                              </p>
                            </div>
                          </a>
                        </div>
                        <div
                          class="value_props50"
                        >
                          <a
                            href="/"
                          >
                            <div
                              class="value_icon"
                            >
                              <img
                                alt=""
                                class="Help-icon-page"
                                src="CourseManagement.png"
                              />
                            </div>
                            <div
                              class="value_content"
                            >
                              <h4>
                                Course Management
                              </h4>
                              <p>
                                Maintain your course and engage with students.
                              </p>
                            </div>
                          </a>
                        </div>
                        <div
                          class="value_props50"
                        >
                          <a
                            href="/"
                          >
                            <div
                              class="value_icon"
                            >
                              <img
                                alt=""
                                class="Help-icon-page"
                                src="TrueFile.png"
                              />
                            </div>
                            <div
                              class="value_content"
                            >
                              <h4>
                                Trust & Safety
                              </h4>
                              <p>
                                Policy and copyright questions and guidance.
                              </p>
                            </div>
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="tpc152"
                  >
                    <div
                      class="crse_content"
                    >
                      <h3>
                        Frequently Asked Questions
                      </h3>
                    </div>
                    <div
                      class="section3126 mt-20"
                    >
                      <div
                        class="Help2-grid-container"
                      >
                        <a
                          class="value_props51"
                          href="#"
                        >
                          Promote Your Course With Coupons and Referral Links
                        </a>
                        <a
                          class="value_props51"
                          href="#"
                        >
                          Cursus Course Quality Checklist
                        </a>
                        <a
                          class="value_props51"
                          href="#"
                        >
                          Instructor Revenue Share
                        </a>
                        <a
                          class="value_props51"
                          href="#"
                        >
                          Instructor Promotional Agreements and Cursus Deals
                        </a>
                        <a
                          class="value_props51"
                          href="#"
                        >
                          How to Become an Instructor: FAQ
                        </a>
                        <a
                          class="value_props51"
                          href="#"
                        >
                          How to Select Your Payout Method And Become a Premium Instructor
                        </a>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
